src/borg/helpers.py

```diff
@@ -1546,3 +1548,33 @@ def signal_handler(signo, handler):
         yield
     finally:
         signal.signal(signo, old_signal_handler)
+
+
+class ErrorIgnoringTextIOWrapper(io.TextIOWrapper):
+    def read(self, n):
+        if not self.closed:
+            try:
+                return super(ErrorIgnoringTextIOWrapper, self).read(n)
+            except OSError as err:
+                if err.errno != errno.EPIPE:
+                    raise
+                try:
+                    super(ErrorIgnoringTextIOWrapper, self).close()
+                except OSError as err:
+                    if err.errno != errno.EPIPE:
```

Comment:
I'm not sure. It's probably be safer just to ignore any OSError when closing anyway, the only reason I'm closing is to clean up. I've changed this.